File: add/codesnippet/CSharp/m-system.drawing.toolbox_1_1.cs
        private Image GetImageOfCustomControl(Control userControl)
        {
            Image controlImage = null;
            AttributeCollection attrCol = 
                    TypeDescriptor.GetAttributes(userControl);
            ToolboxBitmapAttribute imageAttr = (ToolboxBitmapAttribute)
                attrCol[typeof(ToolboxBitmapAttribute)];
            if (imageAttr != null)
            {
                controlImage = imageAttr.GetImage(userControl);
            }

            return controlImage;
        }

File: add/codesnippet/CPP/m-system.drawing.toolbox_1_1.cpp
private:
    static Image^ GetImageOfCustomControl(Control^ userControl)
    {
        Image^ controlImage = nullptr;
        AttributeCollection^ attrCol =
            TypeDescriptor::GetAttributes(userControl);
        ToolboxBitmapAttribute^ imageAttr = (ToolboxBitmapAttribute^)
            attrCol[ToolboxBitmapAttribute::typeid];
        if (imageAttr != nullptr)
        {
            controlImage = imageAttr->GetImage(userControl);
        }

        return controlImage;
    }